packages/adobe-sign/src/apis/adobe-sign-api.ts

```diff
@@ -1,19 +1,58 @@
-import { AwsSecretKey, AwsSecretName, SecretsManagerService } from "asu-core";
+import {AwsSecretKey, AwsSecretName, AxiosProvider, SecretsManagerService} from "asu-core";
 import { injectable } from "tsyringe";
 import { Agreement } from "../models/agreement";
+import {Axios, AxiosError} from "axios";
+import {ApiError} from "./errors/api-error";
+import {AdobeApiError} from "./errors/adobe-api-error";
+import {InternalApiError} from "./errors/internal-api-error";
 
 @injectable()
 export class AdobeSignApi {
-    constructor(private secretsManagerService: SecretsManagerService) {}
+    private httpClient: Axios;
+    constructor(private secretsManagerService: SecretsManagerService, private axiosProvider: AxiosProvider) {}
+
+    private async getHttpClient(): Promise<Axios> {
+        if (!this.httpClient) {
+            try {
+                console.log('Fetching integration key from secrets manager');
```

Comment:
The two logs in here seem a little verbose. Do you think we will want these logged in prod? Or is this just for debugging?